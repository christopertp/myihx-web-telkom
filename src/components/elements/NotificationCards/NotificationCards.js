import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import styles from './styles.scoped.css';

export default function NotificationCards(props) {
  const { isLoading, notif } = props;
  const { description, status, title } = notif;

  const iconWarning = 'warning';

  const icons = {
    OUTSTANDING: iconWarning,
    ISOLIR: iconWarning,
  };

  const tos = {
    OUTSTANDING: '/bill',
    ISOLIR: '/bill',
  };

  if (!tos[status]) {
    return null;
  }

  const Content = (() => {
    if (isLoading) {
      return <div className="loading" />;
    }

    return (
      <Link to={tos[status]}>
        <img alt="" src={`/assets/grfx_${icons[status]}.svg`} />
        <div>
          <h4>{title}</h4>
          <p>{description}</p>
        </div>
        <img alt="" src="/assets/ic_chevron_white.svg" />
      </Link>
    );
  })();

  return (
    <section className={styles.root}>
      {Content}
    </section>
  );
}

NotificationCards.defaultProps = {
  isLoading: false,
  notif: {},
};

NotificationCards.propTypes = {
  isLoading: PropTypes.bool,
  notif: PropTypes.shape({
    description: PropTypes.string,
    status: PropTypes.string,
    title: PropTypes.string,
  }),
};
