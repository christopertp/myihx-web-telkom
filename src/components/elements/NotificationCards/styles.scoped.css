.root {
  margin: 2rem 0 0;
}

.root > a, .root > div {
  border-radius: 0.5rem;
  margin-bottom: 1rem;
}

.root > div {
  height: 6.625rem;
  width: 25rem;
}

.root > a {
  align-items: center;
  background: url('/assets/bg_banner_plain.svg') center center no-repeat;
  background-size: cover;
  box-shadow: 0 0.5rem 1rem rgba(0, 0, 1, 0.15);
  display: inline-flex;
  height: 6.5rem;
  padding: 1rem;
  width: 31.8rem;
}

.root > a > img:first-of-type { height: 4rem; }
.root > a > img:last-of-type { height: 1.5rem; }

.root > a > div {
  color: #ffffff;
  flex: 1;
  margin: 0 1rem;
  white-space: normal;
}

.root > a > div > small {
  opacity: 0.7;
  font-size: 0.75rem;
  font-weight: 800;
}

.root > a > div > h4 {
  font-size: 1rem;
  margin: 0.25rem 0;
}

.root > a > div > p { font-size: 0.75rem; }

@media only screen and (max-width: 48em) {
  .root > h4 { font-size: 0.75rem; }
  .root > a, .root > div {
    margin-bottom: 0.25rem;
    width: 100%;
  }
  .root > a > img:first-of-type { height: 3rem; }
}
