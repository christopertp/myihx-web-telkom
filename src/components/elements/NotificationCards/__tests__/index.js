import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import NotificationCards from '../NotificationCards';

describe('src/components/elements/NotificationCards', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const data = { title: 'Sorry, you have an outstanding bill', description: 'In order to use your service, please complete the payment', status: 'OUTSTANDING' };
    const tree = shallow.render(<NotificationCards notif={data} />);
    expect(tree).toMatchSnapshot();
  });

  test('render loading', () => {
    const notif = { status: 'OUTSTANDING' };
    const isLoading = true;
    const result = NotificationCards({ notif, isLoading });
    expect(result.props.children.props.className).toBe('loading');
  });

  test('render no status', () => {
    const notif = { status: '' };
    const result = NotificationCards({ notif, isLoading: false });
    expect(result).toBe(null);
  });
});
