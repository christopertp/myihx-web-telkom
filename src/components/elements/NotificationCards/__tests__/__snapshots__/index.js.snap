// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/elements/NotificationCards render 1`] = `
<section>
  <Link
    to="/bill"
  >
    <img
      alt=""
      src="/assets/grfx_warning.svg"
    />
    <div>
      <h4>
        Sorry, you have an outstanding bill
      </h4>
      <p>
        In order to use your service, please complete the payment
      </p>
    </div>
    <img
      alt=""
      src="/assets/ic_chevron_white.svg"
    />
  </Link>
</section>
`;
